tree/ntuple/v7/src/RNTupleDescriptor.cxx

```diff
@@ -102,6 +104,25 @@ ROOT::Experimental::RFieldDescriptor::CreateField(const RNTupleDescriptor &ntplD
    return field;
 }
 
+bool ROOT::Experimental::RFieldDescriptor::IsCustomClass() const
+{
+   if (fStructure != ENTupleStructure::kRecord && fStructure != ENTupleStructure::kUnsplit)
+      return false;
+
+   // Skip untyped structs
+   if (fTypeName.empty())
+      return false;
+
+   if (fStructure == ENTupleStructure::kRecord) {
```

Comment:
Are `std::optional` and `std::variant` "custom" or which part of the code is making this 'false'?

Reply:
`std::optional` is internally stored as a collection (of zero or one elements), `std::variant` has its own structural role (`ENTupleStructure::kVariant`).